### PR TITLE
New documentation sections in the IO module

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Resolves: https://github.com/chapel-lang/chapel/issues/19529

Adds a few new documentation sections to the IO module:
* `I/O Transactions`: about mark/commit/revert and buffering behavior
* `Specifying the region of a FileReader or FileWriter`
* `Locking Behavior of FileReaders and FileWriters`: about the `locking` param and manual lock/unlock

Each section links to relevant procedures, and some procedures link back to those sections. A new test is added to match one of the new examples included in the docs.

- [x] paratest
- [x] inspected built docs

### CODE REVIEW

File: test/library/standard/IO/readConcurrent.compopts [hunk 1]
```diff
@@ -0,0 +1 @@
+-sReadBinaryArrayReturnInt=true -suseNewFileReaderRegionBounds=true
```
Comment (lydia-duncan) [inline test/library/standard/IO/readConcurrent.compopts:1]: The fileReaderRegionBounds argument will generate a deprecation message on main, you'll want to merge with main and re-run this test before merging

**Review by lydia-duncan — COMMENTED** [review]
Just some clean ups that you'll want to address before merging this.  Thanks for taking this on, the documentation is much improved with your work here :)

File: modules/standard/IO.chpl [hunk 2]
```diff
@@ -456,6 +456,171 @@ recovered from separately from a ``SystemError``. See the following example:
...
+
+When a ``fileReader`` or ``fileWriter`` is initialized using one of the
+following routines, the optional ``region`` argument can be set to designate
+some region of the file (a zero-based :ref:`range<Chapter-Ranges>` of integers
+in bytes) that can be read from or written to:
+
+* :proc:`file.reader`
+* :proc:`file.writer`
+* :proc:`openReader`
+* :proc:`openWriter`
```
[the remark this answers is not in the document]
Comment (lydia-duncan) [reply modules/standard/IO.chpl]: I was hoping we'd be able to remove the style version a lot sooner, which is why it got left like that 😅

[the remark this answers is not in the document]
Comment (jeremiah-corrado) [reply modules/standard/IO.chpl]: Oh I see. I'll remove the link. Thanks!

**Review by lydia-duncan — APPROVED** [review]
Thanks! (I didn't look through the built docs again, it looked like you were handling that)

**Pull request merged**